### ISSUE TITLE
Delete executions per page instead of batching them

### ISSUE THREAD

**Issue by ctrlrsf** [phwitservices/RundeckLogfileCleanup#3]
Hi,

Thanks for this script, very helpful!

While using it on our system we found it was less impactful to delete the executions per page instead of batching them all up and doing one delete per. In this case we end up making a few more delete API calls, but it allows for the number of executions to start reducing as soon as the pages are returned.

For example, if we have 50 pages of 1000 jobs each, we wouldn't start deleting the 50000 jobs until all 50 pages returned. With the proposed change, we would start chipping away at the 50000 jobs much earlier on. If we stopped the job half way, next time its ran it would only have 25000 jobs to work through.

Would you be interested in a PR to switch to this behavior?

**Comment by phwitservices** [comment]
Sorry for the late response to this, and thanks for your interest.

One version of this script deleted executions as soon as they were returned. There is a problem which I think your suggestion hits. What I found was that deleting as I went along altered the get_executions_per_job() paging so that some executions were never returned.

For instance, suppose we had 3 executions all of which were old, and we'd set the pagesize to 1.The first call to get_exec... would see jobs (1,2,3) and return (1) which we'd then delete. The next call would see executions (2,3) and by asking for page 2 we'd get execution (3). We'd never see the 2nd execution.

Once I spotted this I switched to storing all the deletes needed and pulling the entire job history before doing any deletes.
